lib/pbench/test/unit/server/query_apis/test_datasets_update.py

```diff
@@ -228,10 +228,128 @@ def fake_bulk(
         monkeypatch.setattr("elasticsearch.helpers.streaming_bulk", fake_bulk)
 
         response = client.post(
-            f"{server_config.rest_uri}/datasets/publish/random_md5_string1",
+            f"{server_config.rest_uri}/datasets/random_md5_string1",
             headers={"authorization": f"Bearer {pbench_token}"},
             query_string=self.PAYLOAD,
         )
 
         # Verify the failure
         capinternal("Unexpected backend error", response)
+
+    @pytest.mark.parametrize(
+        "owner",
+        ("drb", "test"),
+    )
+    def test_query_owner_publish(
+        self,
+        attach_dataset,
+        build_auth_header,
+        client,
+        create_drb_user,
+        get_document_map,
+        monkeypatch,
+        owner,
+        server_config,
+    ):
+        """
+        Check behavior of the datasets_update API with various combinations of dataset access and
```

Comment:
Wrapping is bad.

Also, I think this could be "parametrized" to set different combinations of the `owner` and `access` parameters to prove that all the combinations work correctly.

Also, take a look at the `test_requests.py` and `test_datasets_metadata.py` tests: while this doesn't directly relate to your changes it'd be a good "opportunity" to add audit log verification after the update!